rundnsmasq.sh

```diff
@@ -1,28 +1,33 @@
 #!/usr/bin/bash
 
-IP=${IP:-"172.22.0.1"}
+PROVISIONING_INTERFACE=${PROVISIONING_INTERFACE:-"provisioning"}
+
 HTTP_PORT=${HTTP_PORT:-"80"}
 DHCP_RANGE=${DHCP_RANGE:-"172.22.0.10,172.22.0.100"}
-INTERFACE=${INTERFACE:-"provisioning"}
-EXCEPT_INTERFACE=${EXCEPT_INTERFACE:-"lo"}
+DNSMASQ_EXCEPT_INTERFACE=${DNSMASQ_EXCEPT_INTERFACE:-"lo"}
+
+PROVISIONING_IP=`ip -4 address show dev $PROVISIONING_INTERFACE | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | head -n 1`
```

Comment:
please use $() instead of backquotes to wrap the command
wrap $PROVISIONING_INTERFACE in double-quotes "
also I would be careful with perl-regex in grep as it's not universally portable, but it's probably fine